### PR TITLE
refactor(lint): apply clang-tidy fixes

### PR DESCRIPTION
**Pull request by kylo252**
List of most notable fixes:
  - enclose macro arguments in parentheses (bugprone)
  - variable declaration

### CODE REVIEW

File: src/nvim/eval/typval.c [hunk 1]
```diff
@@ -2927,7 +2928,7 @@ bool tv_check_str_or_nr(const typval_T *const tv)
 
 #define FUNC_ERROR "E703: Using a Funcref as a Number"
 
-static const char *const num_errors[] = {
+static const char *const kNumErrors[] = {
```
Comment (justinmk) [inline src/nvim/eval/typval.c:2931]: Not in favor this rename. Why keep sending PRs with this?

**Comment by dundargoc** [comment]
Looks good so far. 2 pointers:
- Don't automatically remove casts even if they seem unnecessary. `clang-tidy` can't account for different architectures and may remove a cast that is unnecessary on the architecture it was run on, but not on others.
- After running clang-tidy, manually "join" together empty declarations without assignment as they only take up space.

So this:

```c
int a, b = 5;
```

should be split up while this

```c
int a, b, c, d, e;
```

should be on one line.

Also, a rebase is needed since there's a conflict.

**Comment by kylo252** [comment]
> Looks good so far. 2 pointers:
> 
> * Don't automatically remove casts even if they seem unnecessary. `clang-tidy` can't account for different architectures and may remove a cast that is unnecessary on the architecture it was run on, but not on others.

The only casts I removed are related to casting a loop variable of type `int` to `int`, when it was declared just a line or two before. But I know what you mean, and I was careful to selectively apply the obvious ones.

**Comment by kylo252** [comment]
Closed in favor of https://github.com/neovim/neovim/pull/17443

**Pull request closed without merging**